### DISCUSSION TITLE
install seurat5 failed.Error: Failed to install 'Seurat' from GitHub:   ! System command 'Rcmd.exe' failed.

### DISCUSSION THREAD

**Discussion by XIANZHE-LI** [satijalab/seurat#7135] [Q&A]
> remotes::install_github("satijalab/seurat", "seurat5", quiet = TRUE)
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

 1: All                                               
 2: CRAN packages only                                
 3: None                                              
 4: SeuratObject (4.1.3      -> 87ea3960c...) [GitHub]
 5: vctrs        (0.5.2      -> 0.6.1       ) [CRAN]  
 6: rlang        (1.0.6      -> 1.1.0       ) [CRAN]  
 7: cli          (3.6.0      -> 3.6.1       ) [CRAN]  
 8: spatstat.... (3.0-1      -> 3.0-2       ) [CRAN]  
 9: spatstat.... (3.0-6      -> 3.1-0       ) [CRAN]  
10: spatstat.... (3.0-0      -> 3.0-1       ) [CRAN]  
11: htmltools    (0.5.4      -> 0.5.5       ) [CRAN]  
12: gtable       (0.3.1      -> 0.3.3       ) [CRAN]  
13: tibble       (3.1.8      -> 3.2.1       ) [CRAN]  
14: pillar       (1.8.1      -> 1.9.0       ) [CRAN]  
15: dplyr        (1.1.0      -> 1.1.1       ) [CRAN]  
16: xfun         (0.37       -> 0.38        ) [CRAN]  
17: rmarkdown    (2.20       -> 2.21        ) [CRAN]  
18: commonmark   (1.8.1      -> 1.9.0       ) [CRAN]  
19: parallelly   (1.34.0     -> 1.35.0      ) [CRAN]  
20: FNN          (1.1.3.1    -> 1.1.3.2     ) [CRAN]  
21: spatstat.... (3.0-0      -> 3.0-1       ) [CRAN]  
22: spatstat.... (3.1-3      -> 3.1-4       ) [CRAN]  
23: RcppArmad... (0.12.0.1.0 -> 0.12.2.0.0  ) [CRAN]  
24: ggplot2      (3.4.1      -> 3.4.2       ) [CRAN]  
25: htmlwidgets  (1.6.1      -> 1.6.2       ) [CRAN]  
26: igraph       (1.4.1      -> 1.4.2       ) [CRAN]  
27: spatstat.... (3.0-6      -> 3.1-0       ) [CRAN]  

Enter one or more numbers, or an empty line to skip updates: 5
Installing 1 packages: vctrs
还安装相依关系‘rlang’

程序包‘rlang’打开成功，MD5和检查也通过
Warning: 无法将拆除原来安装的程序包‘rlang’
Warning: 回复了‘rlang’
程序包‘vctrs’打开成功，MD5和检查也通过
Warning: 无法将拆除原来安装的程序包‘vctrs’
Warning: 回复了‘vctrs’
Error: Failed to install 'Seurat' from GitHub:
  ! System command 'Rcmd.exe' failed
In addition: Warning messages:
1: In file.copy(savedcopy, lib, recursive = TRUE) :
  problem copying C:\Users\Lenovo\AppData\Local\R\win-library\4.2\00LOCK\rlang\libs\x64\rlang.dll to C:\Users\Lenovo\AppData\Local\R\win-library\4.2\rlang\libs\x64\rlang.dll: Permission denied
2: In file.copy(savedcopy, lib, recursive = TRUE) :
  problem copying C:\Users\Lenovo\AppData\Local\R\win-library\4.2\00LOCK\vctrs\libs\x64\vctrs.dll to C:\Users\Lenovo\AppData\Local\R\win-library\4.2\vctrs\libs\x64\vctrs.dll: Permission denied

how can i fix the problem?

**Comment by wang0430** [comment]
我也遇到了这个问题，也太难解决了 ，根本就是没有办法

**Comment by wang0430** [comment]
请问你解决了吗

**Reply by tyhan2360** [reply]
用RGui安装，完了在Rstudio里面使用，亲测有效~~
> library(Seurat)
载入需要的程辑包：SeuratObject
载入需要的程辑包：sp
The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,
which was just loaded, will retire in October 2023.
Please refer to R-spatial evolution reports for details, especially
https://r-spatial.org/r/2023/05/15/evolution4.html.
It may be desirable to make the sf package available;
package maintainers should consider adding sf to Suggests:.
The sp package is now running under evolution status 2
     (status 2 uses the sf package in place of rgdal)

载入程辑包：‘SeuratObject’

The following objects are masked from ‘package:base’:

    intersect, saveRDS

Loading Seurat v5 beta version 
To maintain compatibility with previous workflows, new Seurat objects will use the previous object structure by default
To use new Seurat v5 assays: Please run: options(Seurat.object.assay.version = 'v5')
> packageVersion('Seurat')
[1] ‘4.9.9.9050’

**Reply by XIANZHE-LI** [reply]
> remotes::install_github("satijalab/seurat", "seurat5", quiet = TRUE)
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                          
2: CRAN packages only                           
3: None                                         
4: SeuratObject (4.1.3 -> c51dd865f...) [GitHub]
5: spatstat.... (3.2-1 -> 3.2-4       ) [CRAN]  
6: igraph       (1.5.0 -> 1.5.0.1     ) [CRAN]  

Enter one or more numbers, or an empty line to skip updates: 3
错误: Failed to install 'Seurat' from GitHub:
  ! System command 'Rcmd.exe' failed

还是失败了，在rgui4.2.1安装

**Reply by XIANZHE-LI** [reply]
<img width="511" alt="image" src="https://github.com/satijalab/seurat/assets/113411292/bdfee708-47b9-43c0-81e1-85ec9e1f6cca">

**Comment by honghh2018** [comment]
Encounter the same error, had anyone solve this issue?

**Reply by tyhan2360** [reply]
Install by RGui, and then can use in Rstudio.

**Reply by liulu98** [reply]
It's effective! Thanks!

**Comment by XIANZHE-LI** [comment]
安装成功了。将所有安装包移走，除了base包。再安装V5。
<html>
<body>


> remotes::install_github("satijalab/seurat", "seurat5", quiet = TRUE) Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Installing 10 packages: Rcpp, rlang, future, RcppEigen, generics, lifecycle, spam, progressr, future.apply, sp Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format Error in readRDS(file) : unknown input format 程序包‘Rcpp’打开成功，MD5和检查也通过 程序包‘rlang’打开成功，MD5和检查也通过 程序包‘future’打开成功，MD5和检查也通过 程序包‘RcppEigen’打开成功，MD5和检查也通过 程序包‘generics’打开成功，MD5和检查也通过 程序包‘lifecycle’打开成功，MD5和检查也通过 程序包‘spam’打开成功，MD5和检查也通过 程序包‘progressr’打开成功，MD5和检查也通过 程序包‘future.apply’打开成功，MD5和检查也通过 程序包‘sp’打开成功，MD5和检查也通过 Running `R CMD build`... Installing 127 packages: sitmo, BH, Rcpp, rlang, lifecycle, vctrs, utf8, fansi, spatstat.utils, tensor, abind, polyclip, deldir, spatstat.geom, spatstat.data, fastmap, rappdirs, R6, fs, ellipsis, sass, mime, memoise, jsonlite, jquerylib, htmltools, cachem, base64enc, magrittr, later, promises, gtable, stringi, stringr, plyr, colorspace, pkgconfig, pillar, viridisLite, RColorBrewer, munsell, labeling, farver, withr, tibble, scales, isoband, future, tidyselect, generics, bitops, caTools, gtools, rprojroot, lazyeval, cpp11, purrr, dplyr, tinytex, fontawesome, bslib, xfun, highr, evaluate, rmarkdown, knitr, yaml, sys, askpass, openssl, curl, commonmark, crayon, sourcetools, xtable, httpuv, png, here, RcppTOML, dqrng, RcppProgress, irlba, RcppAnnoy, FNN, goftest, spatstat.sparse, spatstat.random, RcppArmadillo, matrixStats, gridExtra, reshape2, ggplot2, future.apply, RcppEigen, gplots, data.table, crosstalk, tidyr, htmlwidgets, httr, shiny, zoo, igraph, reticulate, progressr, sp, uwot, spatstat.explore, sctransform, scattermore, Rtsne, RSpectra, ROCR, RcppHNSW, RANN, plotly, pbapply, patchwork, miniUI, lmtest, leiden, ica, ggridges, ggrepel, fitdistrplus, fastDummies, cowplot    有二进制版本的，但源代码版本是后来的:        binary  source needs_compilation igraph  1.5.0 1.5.0.1              TRUE  程序包‘sitmo’打开成功，MD5和检查也通过 程序包‘BH’打开成功，MD5和检查也通过 程序包‘Rcpp’打开成功，MD5和检查也通过 程序包‘rlang’打开成功，MD5和检查也通过 程序包‘lifecycle’打开成功，MD5和检查也通过 程序包‘vctrs’打开成功，MD5和检查也通过 程序包‘utf8’打开成功，MD5和检查也通过 程序包‘fansi’打开成功，MD5和检查也通过 程序包‘spatstat.utils’打开成功，MD5和检查也通过 程序包‘tensor’打开成功，MD5和检查也通过 程序包‘abind’打开成功，MD5和检查也通过 程序包‘polyclip’打开成功，MD5和检查也通过 程序包‘deldir’打开成功，MD5和检查也通过 程序包‘spatstat.geom’打开成功，MD5和检查也通过 程序包‘spatstat.data’打开成功，MD5和检查也通过 程序包‘fastmap’打开成功，MD5和检查也通过 程序包‘rappdirs’打开成功，MD5和检查也通过 程序包‘R6’打开成功，MD5和检查也通过 程序包‘fs’打开成功，MD5和检查也通过 程序包‘ellipsis’打开成功，MD5和检查也通过 程序包‘sass’打开成功，MD5和检查也通过 程序包‘mime’打开成功，MD5和检查也通过 程序包‘memoise’打开成功，MD5和检查也通过 程序包‘jsonlite’打开成功，MD5和检查也通过 程序包‘jquerylib’打开成功，MD5和检查也通过 程序包‘htmltools’打开成功，MD5和检查也通过 程序包‘cachem’打开成功，MD5和检查也通过 程序包‘base64enc’打开成功，MD5和检查也通过 程序包‘magrittr’打开成功，MD5和检查也通过 程序包‘later’打开成功，MD5和检查也通过 程序包‘promises’打开成功，MD5和检查也通过 程序包‘gtable’打开成功，MD5和检查也通过 程序包‘stringi’打开成功，MD5和检查也通过 程序包‘stringr’打开成功，MD5和检查也通过 程序包‘plyr’打开成功，MD5和检查也通过 程序包‘colorspace’打开成功，MD5和检查也通过 程序包‘pkgconfig’打开成功，MD5和检查也通过 程序包‘pillar’打开成功，MD5和检查也通过 程序包‘viridisLite’打开成功，MD5和检查也通过 程序包‘RColorBrewer’打开成功，MD5和检查也通过 程序包‘munsell’打开成功，MD5和检查也通过 程序包‘labeling’打开成功，MD5和检查也通过 程序包‘farver’打开成功，MD5和检查也通过 程序包‘withr’打开成功，MD5和检查也通过 程序包‘tibble’打开成功，MD5和检查也通过 程序包‘scales’打开成功，MD5和检查也通过 程序包‘isoband’打开成功，MD5和检查也通过 程序包‘future’打开成功，MD5和检查也通过 程序包‘tidyselect’打开成功，MD5和检查也通过 程序包‘generics’打开成功，MD5和检查也通过 程序包‘bitops’打开成功，MD5和检查也通过 程序包‘caTools’打开成功，MD5和检查也通过 程序包‘gtools’打开成功，MD5和检查也通过 程序包‘rprojroot’打开成功，MD5和检查也通过 程序包‘lazyeval’打开成功，MD5和检查也通过 程序包‘cpp11’打开成功，MD5和检查也通过 程序包‘purrr’打开成功，MD5和检查也通过 程序包‘dplyr’打开成功，MD5和检查也通过 程序包‘tinytex’打开成功，MD5和检查也通过 程序包‘fontawesome’打开成功，MD5和检查也通过 程序包‘bslib’打开成功，MD5和检查也通过 程序包‘xfun’打开成功，MD5和检查也通过 程序包‘highr’打开成功，MD5和检查也通过 程序包‘evaluate’打开成功，MD5和检查也通过 程序包‘rmarkdown’打开成功，MD5和检查也通过 程序包‘knitr’打开成功，MD5和检查也通过 程序包‘yaml’打开成功，MD5和检查也通过 程序包‘sys’打开成功，MD5和检查也通过 程序包‘askpass’打开成功，MD5和检查也通过 程序包‘openssl’打开成功，MD5和检查也通过 程序包‘curl’打开成功，MD5和检查也通过 程序包‘commonmark’打开成功，MD5和检查也通过 程序包‘crayon’打开成功，MD5和检查也通过 程序包‘sourcetools’打开成功，MD5和检查也通过 程序包‘xtable’打开成功，MD5和检查也通过 程序包‘httpuv’打开成功，MD5和检查也通过 程序包‘png’打开成功，MD5和检查也通过 程序包‘here’打开成功，MD5和检查也通过 程序包‘RcppTOML’打开成功，MD5和检查也通过 程序包‘dqrng’打开成功，MD5和检查也通过 程序包‘RcppProgress’打开成功，MD5和检查也通过 程序包‘irlba’打开成功，MD5和检查也通过 程序包‘RcppAnnoy’打开成功，MD5和检查也通过 程序包‘FNN’打开成功，MD5和检查也通过 程序包‘goftest’打开成功，MD5和检查也通过 程序包‘spatstat.sparse’打开成功，MD5和检查也通过 程序包‘spatstat.random’打开成功，MD5和检查也通过 程序包‘RcppArmadillo’打开成功，MD5和检查也通过 程序包‘matrixStats’打开成功，MD5和检查也通过 程序包‘gridExtra’打开成功，MD5和检查也通过 程序包‘reshape2’打开成功，MD5和检查也通过 程序包‘ggplot2’打开成功，MD5和检查也通过 程序包‘future.apply’打开成功，MD5和检查也通过 程序包‘RcppEigen’打开成功，MD5和检查也通过 程序包‘gplots’打开成功，MD5和检查也通过 程序包‘data.table’打开成功，MD5和检查也通过 程序包‘crosstalk’打开成功，MD5和检查也通过 程序包‘tidyr’打开成功，MD5和检查也通过 程序包‘htmlwidgets’打开成功，MD5和检查也通过 程序包‘httr’打开成功，MD5和检查也通过 程序包‘shiny’打开成功，MD5和检查也通过 程序包‘zoo’打开成功，MD5和检查也通过 程序包‘reticulate’打开成功，MD5和检查也通过 程序包‘progressr’打开成功，MD5和检查也通过 程序包‘sp’打开成功，MD5和检查也通过 程序包‘uwot’打开成功，MD5和检查也通过 程序包‘spatstat.explore’打开成功，MD5和检查也通过 程序包‘sctransform’打开成功，MD5和检查也通过 程序包‘scattermore’打开成功，MD5和检查也通过 程序包‘Rtsne’打开成功，MD5和检查也通过 程序包‘RSpectra’打开成功，MD5和检查也通过 程序包‘ROCR’打开成功，MD5和检查也通过 程序包‘RcppHNSW’打开成功，MD5和检查也通过 程序包‘RANN’打开成功，MD5和检查也通过 程序包‘plotly’打开成功，MD5和检查也通过 程序包‘pbapply’打开成功，MD5和检查也通过 程序包‘patchwork’打开成功，MD5和检查也通过 程序包‘miniUI’打开成功，MD5和检查也通过 程序包‘lmtest’打开成功，MD5和检查也通过 程序包‘leiden’打开成功，MD5和检查也通过 程序包‘ica’打开成功，MD5和检查也通过 程序包‘ggridges’打开成功，MD5和检查也通过 程序包‘ggrepel’打开成功，MD5和检查也通过 程序包‘fitdistrplus’打开成功，MD5和检查也通过 程序包‘fastDummies’打开成功，MD5和检查也通过 程序包‘cowplot’打开成功，MD5和检查也通过 安装源码包‘igraph’  Running `R CMD build`...
--
 
> | >
>


</body>
</html>


<img width="1280" alt="image" src="https://github.com/satijalab/seurat/assets/113411292/6298e734-86fd-4c4d-bc08-5c8e548eb612">

**Reply by ZING-ZMJ** [reply]
请问将所有安装包移除？是把之前安的包都删掉吗？
May I know anyone else has resolved this issue?

**Comment by XIANZHE-LI** [comment]
对。你可以把安装的包换个文件夹储存，移除原地址所有的包，在安装就成功了，之后把之前的包在复制回来。



予协助是荷，敬颂时祺，春绥夏禧


李贤哲
————————————
tel:+8618604491305
Jilin,China






---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2023年08月23日 16:48 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [satijalab/seurat] install seurat5 failed.Error: Failed to install 'Seurat' from GitHub: ! System command 'Rcmd.exe' failed. (Discussion #7135) |

请问将所有安装包移除？是把之前安的包都删掉吗？
May I know anyone else has resolved this issue?

**Comment by nilanjana1996** [comment]
I am also facing the same error. Anyone solved the issue? 

Thank you!
